docs/modules/agents.mdx

```diff
@@ -21,18 +25,246 @@ Dive deeper into the concepts behind AI agents in this [research article](https:
+
+# Using OpenAI
+llm = ChatModel.from_name("openai:gpt-5-mini-2025-08-07")
+
+# Using Anthropic
+llm = ChatModel.from_name("anthropic:claude-sonnet-4-20250514")
+
+agent = RequirementAgent(
+    llm=llm,
+    # ... other configuration
+)
+```
+
+```ts TypeScript [expandable]
+COMING SOON
+```
+
+</CodeGroup>
+
+**Backend Features:**
+- Unified Interface: Work with different providers using the same API
+- Model Parameters: Configure temperature, max tokens, and other model settings
+- Provider Support: OpenAI, Anthropic, Ollama, Groq, and more
+- Local & Cloud: Support for both local and cloud-hosted models
+
+### 2. Setting the System Prompt
+
+The system prompt defines your agent's behavior, personality, and capabilities. You can configure this through several parameters when initializing an agent:
+
+<CodeGroup>
+
+```py Python [expandable]
+agent = RequirementAgent(
+    llm=llm,
+    role="You are a helpful research assistant specializing in academic papers",
+    instructions=[
+        "Always provide citations for your sources",
+        "Focus on peer-reviewed research when possible",
+        "Explain complex concepts in simple terms"
+    ],
+    notes=[
+        "Be especially careful about medical or legal advice",
+        "If unsure about a fact, acknowledge the uncertainty"
+    ],
+    name="Research Assistant",
+    description="An AI agent that helps with academic research tasks"
+)
+```
+
+```ts TypeScript [expandable]
+COMING SOON
+```
+
+</CodeGroup>
+
+**Prompt Parameters:**
+- `role`: Defines the agent's persona and primary function
+- `instructions`: List of specific behavioral guidelines 
+- `notes`: Additional context or special considerations
+- `name` and `description`: Help identify the agent's purpose and are helpful when using the `HandoffTool`, `Serve` module, and when you need to access agent metadata via `agent.meta`
 
-**This approach will soon become a default building block for building agents and will replace the others.**
+
+Setting instructions, notes, role will be integrated into the framework provided system prompt template.  If you want to completely override the framework provided system prompt template, you can provide a custom [prompt template](./templates). 
+
+### 3. Configuring Agent Run Options
+When executing an agent, you can provide additional options to guide its behavior and execution settings:
+
+#### Setting Execution Settings and Guiding Agent Run Behavior
+
+<CodeGroup>
+
+```py Python [expandable]
+response = await agent.run(
+    prompt="Analyze the latest AI research trends",
+    expected_output="A structured summary with key findings and recommendations",
+  # expected_output= #insert pydantic data model for structured output
+    backstory="The user is preparing for a conference presentation on AI trends",
+    total_max_retries=5,
+    max_retries_per_step=2,
+    max_iterations=15,
+)
+
+#structured outputs can be accessed like this:
+print(response.output_structured)
```

Comment:
Mention that `output_structured` is defined only when `expected_output` is a Pydantic model or a JSON schema. However, the text representation is always available via `response.output`